src/__tests__/formatPkg.test.js

```diff
@@ -377,3 +377,67 @@ describe('test getRepositoryInfo', () => {
     expect(getRepositoryInfo('aaaaaaaa')).toBe(null);
   });
 });
+
+describe('alternative names', () => {
+  test('name not yet ending in .js', () => {
+    const original = {
+      name: 'places',
+      lastPublisher: { name: 'unknown' },
+    };
+    expect(formatPkg(original)._searchInternal.alternativeNames)
+      .toMatchInlineSnapshot(`
+      Array [
+        "places",
+        "places.js",
+      ]
+    `);
+  });
+
+  test('name ending in .js', () => {
+    const original = {
+      name: 'places.js',
+      lastPublisher: { name: 'unknown' },
+    };
+    expect(formatPkg(original)._searchInternal.alternativeNames)
+      .toMatchInlineSnapshot(`
+            Array [
+              "placesjs",
+              "places js",
+              "places",
+              "places.js",
+            ]
+        `);
+  });
+
+  test('scoped package', () => {
+    const original = {
+      name: '@algolia/places.js',
+      lastPublisher: { name: 'unknown' },
+    };
+    expect(formatPkg(original)._searchInternal.alternativeNames)
+      .toMatchInlineSnapshot(`
+            Array [
+              "algoliaplacesjs",
+              " algolia places js",
```

Comment:
maybe a `.trim()` would clean that

Reply:
yes, however it isn't really an issue for the engine, so I left it as-is